bot/code_review_bot/report/base.py

```diff
@@ -97,27 +80,38 @@ def requires(self, configuration, *keys):
     def calc_stats(self, issues):
         """
         Calc stats about issues:
-        * group issues by class name
+        * group issues by analyzer
         * count their total number
         * count their publishable number
         """
-        groups = itertools.groupby(
-            sorted(issues, key=lambda x: str(x.__class__)), lambda x: x.__class__
-        )
 
-        def stats(items):
+        def by_analyzer(i):
+            return i.analyzer
+
+        groups = itertools.groupby(sorted(issues, key=by_analyzer), by_analyzer)
+
+        def stats(analyzer, items):
             _items = list(items)
+
+            # Lookup the help message, supporting regexes
+            _help = HELP_COMMANDS.get(analyzer)
+            if _help is None:
+                for regex, help_value in HELP_COMMANDS.items():
+                    if isinstance(regex, Pattern) and regex.match(analyzer):
+                        _help = help_value
+                        break
+
             return {
+                "analyzer": analyzer,
+                "help": _help,
                 "total": len(_items),
                 "publishable": sum([i.is_publishable() for i in _items]),
                 "publishable_paths": list(
                     {i.path for i in _items if i.is_publishable()}
                 ),
             }
 
-        from collections import OrderedDict
-
-        return OrderedDict([(cls, stats(items)) for cls, items in groups])
+        return [stats(analyzer, items) for analyzer, items in groups]
```

Comment:
Super-nit: return an iterator

Reply:
Or even transform the function into a generator, yielding the stats

Reply:
Not worth it :smiling_imp: